client_test.go

```diff
@@ -117,20 +75,161 @@ func (e *customErrWithCause) Cause() error {
+					Value:      "custom error",
+					Stacktrace: &Stacktrace{Frames: []Frame{}},
+				},
+			},
+		},
+	}
+
+	errorChainTests := []captureExceptionTest{
+		{
+			name: "MostRecentErrorHasStack",
+			err:  pkgErrors.WithStack(&customErr{}),
+			want: []Exception{
+				{
+					Type:  "*sentry.customErr",
+					Value: "wat",
+					// No Stacktrace, because we can't tell where the error came
+					// from and because we have a stack trace in the most recent
+					// error in the chain.
+				},
+				{
+					Type:       "*errors.withStack",
+					Value:      "wat",
+					Stacktrace: &Stacktrace{Frames: []Frame{}},
+				},
+			},
+		},
+		{
+			name: "ChainWithNilCause",
+			err:  &customErrWithCause{cause: nil},
+			want: []Exception{
+				{
+					Type:       "*sentry.customErrWithCause",
+					Value:      "err",
+					Stacktrace: &Stacktrace{Frames: []Frame{}},
+				},
+			},
+		},
+		{
+			name: "ChainWithoutStacktrace",
+			err:  &customErrWithCause{cause: &customErr{}},
+			want: []Exception{
+				{
+					Type:  "*sentry.customErr",
+					Value: "wat",
+				},
+				{
+					Type:       "*sentry.customErrWithCause",
+					Value:      "err",
+					Stacktrace: &Stacktrace{Frames: []Frame{}},
+				},
+			},
+		},
+	}
+
+	tests := []captureExceptionTestGroup{
+		{
+			name:  "Basic",
+			tests: basicTests,
+		},
+		{
+			name:  "ErrorChain",
+			tests: errorChainTests,
+		},
+	}
+
+	for _, grp := range tests {
+		for _, tt := range grp.tests {
+			tt := tt
+			t.Run(grp.name+"/"+tt.name, func(t *testing.T) {
+				client, _, transport := setupClientTest()
+				client.CaptureException(tt.err, nil, nil)
+				if transport.lastEvent == nil {
+					t.Fatal("missing event")
+				}
+				got := transport.lastEvent.Exception
+				if diff := cmp.Diff(tt.want, got); diff != "" {
+					t.Errorf("Event mismatch (-want +got):\n%s", diff)
+				}
+			})
+		}
+	}
+}
+
+func TestCaptureEvent(t *testing.T) {
+	client, _, transport := setupClientTest()
+
+	eventID := EventID("0123456789abcdef")
+	timestamp := time.Now().UTC()
+	serverName := "testServer"
+
+	client.CaptureEvent(&Event{
+		EventID:    eventID,
+		Timestamp:  timestamp,
+		ServerName: serverName,
+	}, nil, nil)
+
+	if transport.lastEvent == nil {
+		t.Fatal("missing event")
+	}
+	want := &Event{
+		EventID:    eventID,
+		Timestamp:  timestamp,
+		ServerName: serverName,
+		Level:      LevelInfo,
+		Platform:   "go",
+		Sdk: SdkInfo{
+			Name:         "sentry.go",
+			Version:      Version,
+			Integrations: []string{},
+			Packages: []SdkPackage{
+				{
+					// FIXME: name format doesn't follow spec in
+					// https://docs.sentry.io/development/sdk-dev/event-payloads/sdk/
+					Name:    "sentry-go",
+					Version: Version,
+				},
+				// TODO: perhaps the list of packages is incomplete or there
+				// should not be any package at all.
+			},
+		},
+	}
+	got := transport.lastEvent
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Errorf("Event mismatch (-want +got):\n%s", diff)
+	}
```

Comment:
This is notably not a very good test, but a starting point to add tests around code that fills in event fields.

It is so that `TestCaptureException` can focus on the list of "exceptions" (errors) instead of dealing with stuff like SDK info, Platform, etc.